lib/active_merchant/billing/gateways/reach.rb

```diff
@@ -97,6 +98,30 @@ def add_customer_data(request, options, payment)
         }.compact
       end
 
+      def add_custom_fields_data(request, options)
+        if options[:fingerprint].present?
+          request[:DeviceFingerprint] = options[:fingerprint]
+          request[:ViaAgent] = false
+        end
+        add_shipping_data(request, options) if options[:consumer_taxes].present?
+        request[:RateOfferId] = options[:rate_offer_id] if options[:rate_offer_id].present?
+        request[:Items] = options[:items] if options[:items].present?
```

Comment:
Would this be a hash with sub-items, or would it only ever be a single string?

Reply:
This would be a hash of sub items